From aebd8edc31e4aeac753a2c8d87edd8c697f14a10 Mon Sep 17 00:00:00 2001
From: js_liangdong <js_liangdong@163.com>
Date: Wed, 21 Apr 2021 16:31:13 +0800
Subject: [PATCH] arm32

---
 tzdata.spec | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/tzdata.spec b/tzdata.spec
index ecf7e5e..8d23016 100644
--- a/tzdata.spec
+++ b/tzdata.spec
@@ -1,6 +1,7 @@
+%global debug_package %{nil}
 Name:		tzdata
 Version:	2019b
-Release:	10
+Release:	11
 Summary:	Timezone data
 License:	Public Domain
 URL:		https://www.iana.org/time-zones
@@ -21,7 +22,9 @@ Patch9005:	remove-Israel-timezone.patch
 BuildRequires:	gawk glibc perl-interpreter
 BuildRequires:	java-devel
 BuildRequires:	glibc-common >= 2.5.90-7
+%ifnarch %{arm}
 BuildArchitectures: noarch
+%endif
 
 %description
 This package contains data files with rules for various timezones around
@@ -118,6 +121,9 @@ install -p -m 644 tzdb.dat $RPM_BUILD_ROOT%{_datadir}/javazi-1.8/
 %{_datadir}/javazi-1.8
 
 %changelog
+* Mon Mar 22 2021 chentt_tt <chen_tingting1@hoperun.com> - 2019b-11
+- add arm32 support
+
 * Mon Mar 23 2020 liuchao<liuchao173@huawei.com> - 2019b-10
 - Type:recommended
 - ID:NA
-- 
2.23.0

