From 10b8cca9eb1dbb394bd89846748d0954e89bdd6e Mon Sep 17 00:00:00 2001
From: root <root@localhost.localdomain>
Date: Mon, 22 Feb 2021 02:33:16 -0500
Subject: [PATCH] glibc-add-arm32-support

---
 glibc.spec | 20 +++++++++++++++-----
 1 file changed, 15 insertions(+), 5 deletions(-)

diff --git a/glibc.spec b/glibc.spec
index 1dc2725..372ffa6 100644
--- a/glibc.spec
+++ b/glibc.spec
@@ -59,7 +59,7 @@
 ##############################################################################
 Name: 	 	glibc
 Version: 	2.28
-Release: 	36
+Release: 	38
 Summary: 	The GNU libc libraries
 License:	%{all_license}
 URL: 		http://www.gnu.org/software/glibc/
@@ -110,7 +110,11 @@ BuildRequires: gcc-c++ libstdc++-static glibc-devel libidn2
 
 Requires: glibc-common = %{version}-%{release}
 Requires: glibc-langpack = %{version}-%{release}
+%ifarch armv7hl
+Requires(pre): base-files
+%else
 Requires(pre): basesystem
+%endif
 
 %description
 The GNU C Library project provides the core libraries for the GNU system and
@@ -528,7 +532,7 @@ pushd build-%{target}
     -I. -DDATADIR=\"%{_datadir}\" -DPREFIX=\"%{_prefix}\" \
     -L../build-%{target} \
     -B../build-%{target}/csu/ -lc -lc_nonshared \
-    -Wl,-dynamic-linker=/lib64/ld-%{version}.so \
+    -Wl,-dynamic-linker=/%{_lib}/ld-%{version}.so \
     -Wl,-rpath-link=.:./math:./elf:./dlfcn:./nss:./rt:./resolv:./mathvec:./support:./nptl libc.so.6 libc_nonshared.a \
     -Wl,--as-needed $olddir/build-%{target}/elf/ld.so
 install -m 700 build-locale-archive $RPM_BUILD_ROOT%{_prefix}/sbin/build-locale-archive
@@ -735,8 +739,8 @@ fi
 %verify(not md5 size mtime) %config(noreplace) /etc/nsswitch.conf
 %verify(not md5 size mtime) %config(noreplace) /etc/ld.so.conf
 %verify(not md5 size mtime) %config(noreplace) /etc/rpc
-%verify(not md5 size mtime) %config(noreplace) /usr/lib64/gconv/gconv-modules
-%verify(not md5 size mtime) /usr/lib64/gconv/gconv-modules.cache
+%verify(not md5 size mtime) %config(noreplace) %{_libdir}/gconv/gconv-modules
+%verify(not md5 size mtime) %{_libdir}/gconv/gconv-modules.cache
 %dir /etc/ld.so.conf.d
 %dir %{_prefix}/libexec/getconf
 %{_prefix}/libexec/getconf/*
@@ -902,7 +906,7 @@ fi
 %{_prefix}/bin/pcprofiledump
 %{_prefix}/bin/xtrace
 %{_prefix}/lib/debug/usr/bin/*.debug
-%{_prefix}/lib/debug/usr/lib64/*.a
+%{_prefix}/lib/debug%{_libdir}/*.a
 
 
 %files help
@@ -918,6 +922,12 @@ fi
 
 
 %changelog
+* Mon Feb 22 2021 chentt_tt <chen_tingting1@hopefun.com> -2.28-38
+- modify the dependent package basesystem to base-files
+
+* Thu Nov 17 2020 js_liangdong<liang_dong@hoperun.com> - 2.28-37
+- add arm32 compilation support
+
 * Thu Mar 19 2020 yuxiangyang<yuxiangyang4@huawei.com> - 2.28-36
 - fix build src.rpm error
 
-- 
2.20.1

