From 906add35c4e4391095c6a9ea341f206ef9ae2fc7 Mon Sep 17 00:00:00 2001
From: root <root@localhost.localdomain>
Date: Mon, 22 Mar 2021 10:55:05 +0800
Subject: [PATCH] support-for-arm32-compilation-on-mock

---
 gcc_secure.spec | 25 +++++++++++++++++--------
 1 file changed, 17 insertions(+), 8 deletions(-)

diff --git a/gcc_secure.spec b/gcc_secure.spec
index a623aa4..cf524dd 100644
--- a/gcc_secure.spec
+++ b/gcc_secure.spec
@@ -1,9 +1,15 @@
+%if 0%{?mock_build}
+%global install_dir /var/tmp
+%else
+%global install_dir /tmp
+%endif
+
 Name:           gcc_secure
 Summary:        Build with gcov
 License:        GPL
 Group:          System/Management
 Version:        1.0
-Release:        0.6
+Release:        0.7
 BuildRoot:      %{_tmppath}/%{name}-%{version}
 #Source:         %{name}-%{version}.tar.bz2
 
@@ -29,11 +35,11 @@ mkdir -p %{buildroot}/opt/needgcov
 %pre
 
 %post
-echo -e '*cc1_options:\n+ %{!D__KERNEL__:%{!nostdlib:%{!nodefaultlibs:%{!fno-stack-protector:%{!fstack-protector-all:-fstack-protector-strong}}}}}' >/tmp/gcc-specs-fs-cc1
+echo -e '*cc1_options:\n+ %{!D__KERNEL__:%{!nostdlib:%{!nodefaultlibs:%{!fno-stack-protector:%{!fstack-protector-all:-fstack-protector-strong}}}}}' >%{install_dir}/gcc-specs-fs-cc1
 
-echo -e '*cc1_options:\n+ %{!r:%{!D__KERNEL__:%{!pie:%{!fpic:%{!fPIC:%{!fpie:%{!fPIE:%{!fno-pic:%{!fno-PIC:%{!fno-pie:%{!fno-PIE:%{!shared:%{!static:%{!nostdlib:%{!nostartfiles:-fPIE}}}}}}}}}}}}}}}' >/tmp/gcc-specs-pie-cc1
+echo -e '*cc1_options:\n+ %{!r:%{!D__KERNEL__:%{!pie:%{!fpic:%{!fPIC:%{!fpie:%{!fPIE:%{!fno-pic:%{!fno-PIC:%{!fno-pie:%{!fno-PIE:%{!shared:%{!static:%{!nostdlib:%{!nostartfiles:-fPIE}}}}}}}}}}}}}}}' >%{install_dir}/gcc-specs-pie-cc1
 
-echo -e '*self_spec:\n+ %{!D__KERNEL__:%{!pie:%{!A:%{!fno-pie:%{!fno-PIE:%{!fno-pic:%{!fno-PIC:%{!shared:%{!static:%{!r:%{!nostdlib:%{!nostartfiles:-pie}}}}}}}}}}}}' >/tmp/gcc-specs-pie-ld
+echo -e '*self_spec:\n+ %{!D__KERNEL__:%{!pie:%{!A:%{!fno-pie:%{!fno-PIE:%{!fno-pic:%{!fno-PIC:%{!shared:%{!static:%{!r:%{!nostdlib:%{!nostartfiles:-pie}}}}}}}}}}}}' >%{install_dir}/gcc-specs-pie-ld
 
 old_gcc=/usr/bin/gcc
 mv $old_gcc $old_gcc"_old"
@@ -68,9 +74,9 @@ if ! cat /.build.command | egrep "\$gcc_secure_exclude" &>/dev/null; then
 
     if [[ x\$rpm_name = "xglibc" ]] || [[ x\$rpm_name = "xcompat-glibc" ]];then
         #glibc supply fs define, can not add fs for glibc self
-        /usr/bin/gcc_old \$sec_opt "\$@" \$fs_opt --specs=/tmp/gcc-specs-pie-cc1 --specs=/tmp/gcc-specs-pie-ld
+        /usr/bin/gcc_old \$sec_opt "\$@" \$fs_opt --specs=%{install_dir}/gcc-specs-pie-cc1 --specs=%{install_dir}/gcc-specs-pie-ld
     else
-        /usr/bin/gcc_old \$sec_opt "\$@" \$fs_opt --specs=/tmp/gcc-specs-pie-cc1 --specs=/tmp/gcc-specs-pie-ld --specs=/tmp/gcc-specs-fs-cc1
+        /usr/bin/gcc_old \$sec_opt "\$@" \$fs_opt --specs=%{install_dir}/gcc-specs-pie-cc1 --specs=%{install_dir}/gcc-specs-pie-ld --specs=%{install_dir}/gcc-specs-fs-cc1
     fi
 
 else
@@ -110,7 +116,7 @@ if ! cat /.build.command | egrep "\$gcc_secure_exclude" &>/dev/null; then
         #libtool use g++ -v for test compile env,if add Wl opt, it will make g++ -v fail
         $old_gplus"_old" "\$@"
     else
-        /usr/bin/g++_old \$sec_opt "\$@" \$fs_opt  --specs=/tmp/gcc-specs-pie-cc1 --specs=/tmp/gcc-specs-pie-ld --specs=/tmp/gcc-specs-fs-cc1
+        /usr/bin/g++_old \$sec_opt "\$@" \$fs_opt  --specs=%{install_dir}/gcc-specs-pie-cc1 --specs=%{install_dir}/gcc-specs-pie-ld --specs=%{install_dir}/gcc-specs-fs-cc1
     fi
 else
     $old_gplus"_old" "\$@"
@@ -146,7 +152,7 @@ if ! cat /.build.command | egrep "\$gcc_secure_exclude" &>/dev/null; then
         fs_opt=\`cat \$configfile| grep fs_opt| awk -F: '{print \$NF}'\`
     fi
 
-    /usr/bin/c++_old \$sec_opt "\$@" \$fs_opt  --specs=/tmp/gcc-specs-pie-cc1 --specs=/tmp/gcc-specs-pie-ld --specs=/tmp/gcc-specs-fs-cc1
+    /usr/bin/c++_old \$sec_opt "\$@" \$fs_opt  --specs=%{install_dir}/gcc-specs-pie-cc1 --specs=%{install_dir}/gcc-specs-pie-ld --specs=%{install_dir}/gcc-specs-fs-cc1
 
 else
     $old_cpp"_old" "\$@"
@@ -172,6 +178,9 @@ rm -rf %{_tmppath}/%{name}-%{version}
 rm -rf $RPM_BUILD_DIR/%{name}-%{version}
 
 %changelog
+* Thu Nov 17 2020 js_liangdong<liang_dong@hoperun.com> - 1.0-0.7
+- support for arm32 compilation on mock
+
 * Tue Feb 18 2020 yanzhihua<yanzhihua4@huawei.com> - 1.0-0.6
 - Type:enhancement
 - ID:NA
-- 
2.27.0

